metrics.go

```diff
@@ -332,6 +336,23 @@ func (s *ScannerMetrics) Merge(other *ScannerMetrics) {
 	}
 	s.ActivePaths = append(s.ActivePaths, other.ActivePaths...)
 	sort.Strings(s.ActivePaths)
+
+	if len(other.ExcessivePrefixes) > 0 {
+		// Merge and remove duplicates
+		merged := make(map[string]struct{}, len(s.ExcessivePrefixes)+len(other.ExcessivePrefixes))
+		for _, prefix := range s.ExcessivePrefixes {
+			merged[prefix] = struct{}{}
+		}
+		// Add other excessive prefixes
+		for _, prefix := range other.ExcessivePrefixes {
+			merged[prefix] = struct{}{}
+		}
+		s.ExcessivePrefixes = make([]string, 0, len(merged))
+		for prefix := range merged {
+			s.ExcessivePrefixes = append(s.ExcessivePrefixes, prefix)
+		}
+		sort.Strings(s.ExcessivePrefixes)
+	}
```

Comment:
Will it be simpler if `ExcessivePrefixes` is a `StringSet` ?

Reply:
Added https://github.com/minio/minio-go/pull/2107

Reply:
Not sure it is worth including for this one.